src/main/app.js

```diff
@@ -96,189 +109,156 @@ fluid.defaults("gpii.app", {
+                    }
                 },
                 events: {
                     onConnected: "{app}.events.onPSPChannelConnected"
                 }
             }
         },
-        psp: {
-            type: "gpii.app.psp",
+        dialogManager: {
+            type: "gpii.app.dialogManager",
             createOnEvent: "onPSPPrerequisitesReady",
             options: {
                 model: {
                     keyedInUserToken: "{app}.model.keyedInUserToken"
-                }
-            }
-        },
-        waitDialog: {
-            type: "gpii.app.waitDialog",
-            createOnEvent: "onPSPPrerequisitesReady",
-            options: {
+                },
                 modelListeners: {
                     "{lifecycleManager}.model.logonChange": {
-                        changePath: "{that}.model.isShown",
-                        value: "{change}.value.inProgress"
+                        func: "{that}.toggle",
+                        args: ["waitDialog", "{change}.value.inProgress"],
+                        excludeSource: "init"
                     }
                 }
             }
         },
-        restartDialog: {
-            type: "gpii.app.dialog.restartDialog",
-            createOnEvent: "onPSPPrerequisitesReady"
-        },
-        settingsBroker: {
-            type: "gpii.app.settingsBroker",
+        surveyManager: {
+            type: "gpii.app.surveyManager",
             createOnEvent: "onPSPPrerequisitesReady",
             options: {
-                model: {
-                    keyedInUserToken: "{app}.model.keyedInUserToken"
-                },
                 listeners: {
-                    "{psp}.events.onSettingAltered": {
-                        listener: "{that}.enqueue"
-                    },
-                    "{psp}.events.onActivePreferenceSetAltered": {
-                        listener: "{that}.clearPendingChanges"
+                    onSurveyRequired: {
+                        func: "{dialogManager}.show",
+                        args: ["survey", "{arguments}.0"]
                     }
                 }
             }
         },
-        tray: {
-            type: "gpii.app.tray",
+        psp: {
+            type: "gpii.app.psp",
             createOnEvent: "onPSPPrerequisitesReady",
             options: {
                 model: {
-                    keyedInUserToken: "{gpii.app}.model.keyedInUserToken",
-                    pendingChanges: "{settingsBroker}.model.pendingChanges"
-                }
-            }
-        },
-        /*
-         * A helper component used as mediator for handling communication
-         * between the PSP and gpiiConnector components.
-         */
-        channelMediator: {
-            type: "fluid.component",
-            createOnEvent: "onPSPPrerequisitesReady",
-            options: {
+                    keyedInUserToken: "{app}.model.keyedInUserToken"
+                },
+                modelListeners: {
```

Comment:
I think this material configuring the PSP has now become too large to be declared inline in the app. I suggest that this binding material should be factored off into an "incomplete grade" named something like "pspInApp". In strict dependency terms that grade ought to stay in this file but it wouldn't be a blunder to put it into psp.js either, see what your taste is